In [2]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt

In [3]:
dataset_hospitals = pd.read_csv("../Data/train.csv")

In [4]:
dataset_hospitals

encounter_id country  patient_id             race  gender      age  \
0            533253     USA       70110        Caucasian  Female  [70-80)   
1            426224     USA    29775006  AfricanAmerican    Male  [50-60)   
2            634063     USA    80729253        Caucasian  Female  [60-70)   
3            890610     USA     2919042  AfricanAmerican    Male  [60-70)   
4            654194     USA    84871971        Caucasian  Female  [70-80)   
...             ...     ...         ...              ...     ...      ...   
71231        660160     USA    24531381  AfricanAmerican  Female  [80-90)   
71232        826429     USA     4663818  AfricanAmerican  Female  [70-80)   
71233        332030     USA    23397147        Caucasian  Female  [60-70)   
71234        757560     USA    52161750        Caucasian    Male  [60-70)   
71235        459757     USA    88410897        Caucasian    Male  [70-80)   

      weight payer_code  outpatient_visits_in_previous_year  \
0          ?          ?                                   0   
1          ?          ?                                   0   
2          ?          ?                                   0   
3          ?         MC                                   0   
4          ?         HM                                   1   
...      ...        ...                                 ...   
71231      ?         MC                                   0   
71232      ?          ?                                   0   
71233      ?          ?                                   0   
71234      ?         BC                                   0   
71235      ?         BC                                   0   

       emergency_visits_in_previous_year  ...  secondary_diagnosis  \
0                                      0  ...                  276   
1                                      0  ...                  785   
2                                      0  ...                  135   
3                                      0  ...                  562   
4                                      0  ...                  599   
...                                  ...  ...                  ...   
71231                                  1  ...                  786   
71232                                  0  ...                  197   
71233                                  2  ...                  491   
71234                                  0  ...                  157   
71235                                  0  ...                  250   

      additional_diagnosis number_diagnoses  glucose_test_result  \
0                      466                8                  NaN   
1                      162                9                  NaN   
2                      250                6                  NaN   
3                      455                5                  NaN   
4                      428                9                  NaN   
...                    ...              ...                  ...   
71231                  250                9                  NaN   
71232                  V66                6                  NaN   
71233                  276                9                 >300   
71234                250.8                5                  NaN   
71235                  401                8                  NaN   

      a1c_test_result change_in_meds_during_hospitalization  \
0                 NaN                                    No   
1                 NaN                                    No   
2                 NaN                                    Ch   
3                 NaN                                    No   
4                 NaN                                    No   
...               ...                                   ...   
71231              >7                                    Ch   
71232              >7                                    Ch   
71233             NaN                                    Ch   
71234             NaN                

Here we can see that the readmitted_binary doesn't always align with the readmitted_multiclass, even if the readmitted_multiclass is different from "No" we can see that the readmitted_binary is still "No" in those cases.

In [5]:
dataset_hospitals.shape

(71236, 31)

The dataset contains 31 attributes and 71236 records, each attribute being a feature analysed for every encounter had with a patient, representing each row.

In [6]:
dataset_hospitals.dtypes

encounter_id                              int64
country                                  object
patient_id                                int64
race                                     object
gender                                   object
age                                      object
weight                                   object
payer_code                               object
outpatient_visits_in_previous_year        int64
emergency_visits_in_previous_year         int64
inpatient_visits_in_previous_year         int64
admission_type                           object
medical_specialty                        object
average_pulse_bpm                         int64
discharge_disposition                    object
admission_source                         object
length_of_stay_in_hospital                int64
number_lab_tests                          int64
non_lab_procedures                        int64
number_of_medications                     int64
primary_diagnosis                       

Seeing the data types of the attributes show us what we can already simplify as, for example, binary variables like: change_in_meds_during_hospitalization, which has the values "No" or "Ch"; prescribed_diabetes_meds	which has values "Yes" or "No".

In [7]:
dataset_hospitals.isna().sum()

encounter_id                                 0
country                                      0
patient_id                                   0
race                                      3554
gender                                       0
age                                       3557
weight                                       0
payer_code                                   0
outpatient_visits_in_previous_year           0
emergency_visits_in_previous_year            0
inpatient_visits_in_previous_year            0
admission_type                            3706
medical_specialty                            0
average_pulse_bpm                            0
discharge_disposition                     2590
admission_source                          4718
length_of_stay_in_hospital                   0
number_lab_tests                             0
non_lab_procedures                           0
number_of_medications                        0
primary_diagnosis                            0
secondary_dia

In [8]:
dataset_hospitals.describe(include="all").T

count unique                 top  \
encounter_id                           71236.0    NaN                 NaN   
country                                  71236      1                 USA   
patient_id                             71236.0    NaN                 NaN   
race                                     67682      6           Caucasian   
gender                                   71236      3              Female   
age                                      67679     10             [70-80)   
weight                                   71236     10                   ?   
payer_code                               71236     18                   ?   
outpatient_visits_in_previous_year     71236.0    NaN                 NaN   
emergency_visits_in_previous_year      71236.0    NaN                 NaN   
inpatient_visits_in_previous_year      71236.0    NaN                 NaN   
admission_type                           67530      7           Emergency   
medical_specialty                        71236     69                   ?   
average_pulse_bpm                      71236.0    NaN                 NaN   
discharge_disposition                    68646     25  Discharged to home   
admission_source                         66518     16      Emergency Room   
length_of_stay_in_hospital             71236.0    NaN                 NaN   
number_lab_tests                       71236.0    NaN                 NaN   
non_lab_procedures                     71236.0    NaN                 NaN   
number_of_medications                  71236.0    NaN                 NaN   
primary_diagnosis                        71236    687                 428   
secondary_diagnosis                      71236    699                 276   
additional_diagnosis                     71236    747                 250   
number_diagnoses                       71236.0    NaN                 NaN   
glucose_test_result                       3688      3                Norm   
a1c_test_result                          11916      3                  >8   
change_in_meds_during_hospitalization    71236      2                  No   
prescribed_diabetes_meds                 71236      2                 Yes   
medication                               71236    303         ['insulin']   
readmitted_binary                        71236      2                  No   
readmitted_multiclass                    71236      3                  No   

                                        freq             mean  \
encounter_id                             NaN    548798.623716   
country                                71236              NaN   
patient_id                               NaN  54302279.330984   
race                                   50693              NaN   
gender                                 38228              NaN   
age                                    17359              NaN   
weight                                 68990              NaN   
payer_code                             28201              NaN   
outpatient_visits_in_previous_year       NaN         0.369588   
emergency_visits_in_previous_year        NaN         0.196249   
inpatient_visits_in_previous_year        NaN         0.640154   
admission_type                         37742              NaN   
medical_specialty                      34922              NaN   
average_pulse_bpm                        NaN        99.611222   
discharge_disposition                  42256              NaN   
admission_source                       40319              NaN   
length_of_stay_in_hospital               NaN         4.391024   
number_lab_tests                         NaN        43.095654   
non_lab_procedures                       NaN         1.340923   
number_of_medications                    NaN        15.995452   
primary_diagnosis                       4776              NaN   
secondary_diagnosis                     4694              NaN   
additional_diagnosis                    8070              NaN   
number_diagnoses

Between the mean and the 50% quantile there's no much difference which doesn't indicate a possibility of outliers in these (...) parameters.

In [9]:
dataset_hospitals.replace("?", np.nan, inplace=True)

In [10]:
dataset_hospitals.isna().sum() # o numero de nans aumentou o que significa que tinhamos features a "?"

encounter_id                                 0
country                                      0
patient_id                                   0
race                                      5070
gender                                       0
age                                       3557
weight                                   68990
payer_code                               28201
outpatient_visits_in_previous_year           0
emergency_visits_in_previous_year            0
inpatient_visits_in_previous_year            0
admission_type                            3706
medical_specialty                        34922
average_pulse_bpm                            0
discharge_disposition                     2590
admission_source                          4718
length_of_stay_in_hospital                   0
number_lab_tests                             0
non_lab_procedures                           0
number_of_medications                        0
primary_diagnosis                           16
secondary_dia

In [14]:
dataset_hospitals["change_in_meds_during_hospitalization"] = dataset_hospitals["change_in_meds_during_hospitalization"].replace({"No":0, "Ch":1})
dataset_hospitals["prescribed_diabetes_meds"] = dataset_hospitals["prescribed_diabetes_meds"].replace({"No":0, "Yes":1})

In [15]:
dataset_hospitals

encounter_id country  patient_id             race  gender      age  \
0            533253     USA       70110        Caucasian  Female  [70-80)   
1            426224     USA    29775006  AfricanAmerican    Male  [50-60)   
2            634063     USA    80729253        Caucasian  Female  [60-70)   
3            890610     USA     2919042  AfricanAmerican    Male  [60-70)   
4            654194     USA    84871971        Caucasian  Female  [70-80)   
...             ...     ...         ...              ...     ...      ...   
71231        660160     USA    24531381  AfricanAmerican  Female  [80-90)   
71232        826429     USA     4663818  AfricanAmerican  Female  [70-80)   
71233        332030     USA    23397147        Caucasian  Female  [60-70)   
71234        757560     USA    52161750        Caucasian    Male  [60-70)   
71235        459757     USA    88410897        Caucasian    Male  [70-80)   

      weight payer_code  outpatient_visits_in_previous_year  \
0        NaN        NaN                                   0   
1        NaN        NaN                                   0   
2        NaN        NaN                                   0   
3        NaN         MC                                   0   
4        NaN         HM                                   1   
...      ...        ...                                 ...   
71231    NaN         MC                                   0   
71232    NaN        NaN                                   0   
71233    NaN        NaN                                   0   
71234    NaN         BC                                   0   
71235    NaN         BC                                   0   

       emergency_visits_in_previous_year  ...  secondary_diagnosis  \
0                                      0  ...                  276   
1                                      0  ...                  785   
2                                      0  ...                  135   
3                                      0  ...                  562   
4                                      0  ...                  599   
...                                  ...  ...                  ...   
71231                                  1  ...                  786   
71232                                  0  ...                  197   
71233                                  2  ...                  491   
71234                                  0  ...                  157   
71235                                  0  ...                  250   

      additional_diagnosis number_diagnoses  glucose_test_result  \
0                      466                8                  NaN   
1                      162                9                  NaN   
2                      250                6                  NaN   
3                      455                5                  NaN   
4                      428                9                  NaN   
...                    ...              ...                  ...   
71231                  250                9                  NaN   
71232                  V66                6                  NaN   
71233                  276                9                 >300   
71234                250.8                5                  NaN   
71235                  401                8                  NaN   

      a1c_test_result change_in_meds_during_hospitalization  \
0                 NaN                                     0   
1                 NaN                                     0   
2                 NaN                                     1   
3                 NaN                                     0   
4                 NaN                                     0   
...               ...                                   ...   
71231              >7                                     1   
71232              >7                                     1   
71233             NaN                                     1   
71234             NaN                

In [16]:
dataset_hospitals.admission_type

0        Emergency
1        Emergency
2              NaN
3        Emergency
4         Elective
           ...    
71231     Elective
71232       Urgent
71233          NaN
71234    Emergency
71235       Urgent
Name: admission_type, Length: 71236, dtype: object